app/configurator/components/chart-options-selector.tsx

```diff
@@ -42,8 +47,27 @@ export const ChartOptionsSelector = ({
   state: ConfiguratorStateConfiguringChart;
 }) => {
   const locale = useLocale();
-  const [{ data }] = useDataCubeMetadataWithComponentValuesQuery({
-    variables: { iri: state.dataSet, locale },
+  const measures =
+    "y" in state.chartConfig.fields
+      ? [state.chartConfig.fields.y.componentIri]
+      : isTableConfig(state.chartConfig)
+      ? Object.values(state.chartConfig.fields).flatMap((f) =>
+          f.componentType === "Measure" && !f.isHidden ? [f.componentIri] : []
+        )
+      : [];
+
+  const [{ data }] = useDataCubeObservationsQuery({
+    variables: {
+      locale,
+      iri: state.dataSet,
+      measures,
+      filters: state.chartConfig.filters,
+    },
+  });
+
+  const imputationNeeded = useImputationNeeded({
+    chartConfig: state.chartConfig,
+    data: data?.dataCubeByIri?.observations.data,
```

Comment:
Also not sure if this should be done there, but I didn't come up with more elegant solution :)

Reply:
I think it's OK for now :)